### PR TITLE
Shard input for distribution strategy.

### PR DESCRIPTION
**Pull request by dubey**

### CODE REVIEW

**Comment by dubey** [comment]
@yuefengz PTAL.  I added the input_context to the high level input_fn.

File: official/resnet/resnet_run_loop.py [hunk 1]
```diff
@@ -602,7 +602,7 @@ def resnet_main(
       model_dir=flags_obj.model_dir,
       batch_size=flags_obj.batch_size)
 
-  def input_fn_train(num_epochs):
+  def input_fn_train(num_epochs, input_context=None):
```
Comment (yuefengz) [inline official/resnet/resnet_run_loop.py]: Looks like the lambda function in line 633 is not able to accept the input_context?

Comment (dubey) [reply official/resnet/resnet_run_loop.py]: Good catch!  I've fixed this now.

File: official/resnet/resnet_run_loop.py [hunk 2]
```diff
@@ -626,10 +627,13 @@ def input_fn_eval():
                   flags_obj.train_epochs)
 
   use_train_and_evaluate = flags_obj.use_train_and_evaluate or (
-      distribution_strategy.__class__.__name__ == 'CollectiveAllReduceStrategy')
+      distribution_strategy.__class__.__name__ in [
+          'CollectiveAllReduceStrategy', 'MultiWorkerMirroredStrategy'])
   if use_train_and_evaluate:
     train_spec = tf.estimator.TrainSpec(
-        input_fn=lambda: input_fn_train(train_epochs), hooks=train_hooks,
+        input_fn=lambda input_context=None: input_fn_train(
```
[the remark this answers is not in the document]
Comment (yuefengz) [reply official/resnet/resnet_run_loop.py]: NVM, I just tried on ipython and looks like it should work.

**Pull request merged**